lisa/sut_orchestrator/azure/features.py

```diff
@@ -668,6 +669,67 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
             len(all_nics) - self.origin_extra_synthetic_nics_count - 1
         )
 
+    def create_route_table(
+        self,
+        nic_name: str,
+        route_name: str,
+        subnet_mask: str,
+        dest_hop: str,
+        em_first_hop: str = "",
+        next_hop_type: str = "",
+    ) -> None:
+        # some quick checks that the subnet mask looks correct
+        assert_that(subnet_mask).described_as(
+            "subnet mask should be prefix format X.X.X.X/YY"
+        ).contains("/")
+        assert_that(subnet_mask.split("/")).described_as(
+            "subnet mask should be prefix format X.X.X.X/YY"
+        ).is_length(2)
+
+        azure_platform: AzurePlatform = self._platform  # type: ignore
+        # Step 1: create the route table, apply comes later.
+        route_table = self._do_create_route_table(
+            em_first_hop,
+            subnet_mask=subnet_mask,
+            nic_name=nic_name,
+            route_name=route_name,
+            next_hop_type=next_hop_type,
+            dest_hop=dest_hop,
+        )
+        # Step 2: Get the virtual networks in the resource group.
+        vnets: Dict[str, List[str]] = get_virtual_networks(
+            azure_platform, self._resource_group_name
+        )
+        # get the subnets in each virtual network
```

Comment:
Below logic is not like to get for "each" virtual network. It gets first and break.

Reply:
Correct, I'll change the comment and assert if the 'single virtual net' assumption is incorrect.